data-hibernate-jpa/src/test/java/io/micronaut/data/hibernate/querygroupby/MicronautTaskRepository.java

```diff
@@ -0,0 +1,16 @@
+package io.micronaut.data.hibernate.querygroupby;
+
+import io.micronaut.data.annotation.Query;
+import io.micronaut.data.annotation.Repository;
+import io.micronaut.data.hibernate.entities.MicronautTask;
+import io.micronaut.data.repository.CrudRepository;
+
+import java.util.List;
+@Repository
+public interface MicronautTaskRepository extends CrudRepository<MicronautTask, Long> {
+    @Query("select count(*), year(t.dueDate) from MicronautTask t group by year(t.dueDate)")
+    Iterable<TasksPerYear> countByDueYear();
+
+    @Query("select count(*), year(t.dueDate) from MicronautTask t group by year(t.dueDate)")
+    List<List<Integer>> countByDueYearReturnList();
```

Comment:
And this fails with the error 
```
Cannot create TypedQuery for query with more than one return using requested result type [java.util.List]
```
I tried to change it to List<Object[]> which I know Hibernate can return but does not work in our code and we probably can't support it now without some refactoring.